deployment/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/VertxFileDownloader.java

```diff
@@ -51,6 +51,7 @@ public void download(String downloadUrl, String destination, String userName, St
                 Files.deleteIfExists(destinationPath);
                 final AsyncFile destinationFile = vertx.fileSystem().openBlocking(destination, new OpenOptions());
                 final Future<HttpResponse<Void>> future = webClient.getAbs(downloadUrl)
+                        .basicAuthentication(userName, password)
```

Comment:
What happens is username and password are null?   99% of the time I don't think we will have username and password for the public NPM registries?

Reply:
You are right. I had a different solution where I implemented a simple check and removed that in favor of the HttpRequest.basicAuthentication() method. It'd throw a CredentialValidationException. I'll fix it

Reply:
Good catch!